libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -145,21 +152,24 @@ void ManagedQuery::submit_write() {
     query_->submit();
 }
 
-std::shared_ptr<ArrayBuffers> ManagedQuery::submit_read() {
+void ManagedQuery::submit_read() {
+    query_submitted_ = true;
+    query_future_ = std::async(std::launch::async, [&]() {
+        LOG_DEBUG("[ManagedQuery] submit thread start");
+        query_->submit();
+        LOG_DEBUG("[ManagedQuery] submit thread done");
+    });
+}
+
+std::shared_ptr<ArrayBuffers> ManagedQuery::results() {
     if (is_empty_query()) {
         return buffers_;
     }
 
-    query_->submit();
-
-    // Poll status until query is not INPROGRESS
-    Query::Status status;
-    do {
-        status = query_->query_status();
-    } while (status == Query::Status::INPROGRESS);
+    LOG_DEBUG(fmt::format("[ManagedQuery] [{}] Waiting for query", name_));
+    query_future_.wait();
```

Comment:
We should check if the `query_future_` is valid like we do in `ManagedQuery::close()`. 

Something like this:
```c++
if (query_future_.valid()) {
    LOG_DEBUG(fmt::format("[ManagedQuery] [{}] Waiting for query", name_));
    query_future_.wait();
} else {
    // throw error
}
```

We expect it to be valid, but it's good to check.